docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,213 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+Async programming in C# and VB share a language-level asynchronous programming model which allows for easily writing asynchronous code without having to juggle callbacks or conform to a library which supports asynchronicity.
+
+The core of it all are the ``Task`` and ``Task<T>`` objects, which model asynchronous operations.  The async and await keywords provide a natural and synchronous developer experience for interacting with tasks.  The result is the ability to write powerful code which cleanly expresses intent.  There are other ways to approach this than ``async`` and ``await``, outlined in the `Task-based Asynchronous Pattern (TAP) <https://msdn.microsoft.com/en-us/library/hh873175(v=vs.110).aspx>`_, which presents Task-based asynchronocity with options other than the language-level abstractions.  This document will focus on the language-level constructs from this point forward.
```

Comment:
"The core of it all" is a bit vague. You might say "The core of async" ???

You could say types instead of objects in the first sentence to make it super clear that you are talking about the definition. You are talking about modeling, so it kinda aligns. This one is more subtle, since there isn't really any loss of comprehension as currently written.